mvc/controllers/areas.rst

```diff
@@ -1,8 +1,132 @@
 
-.. _issue: https://github.com/aspnet/Docs/issues/121
+This pattern works well for many Web apps. However, for larger apps, this folder hierarchy can become difficult to manage as you have controllers in one parent folder, and views in a separate parent folder. This is where `Areas` come in. Areas enable you to partition your project such that controllers and their views can be stored under a parent folder that defines their logical grouping.
+
+Let's take a look at an example to illustrate how Areas are created and used. In this example, we'll take a simple store app (MyStore) that has a several logical groupings of controllers and actions. These groupings are ``Services``, ``Products``, and ``Deals``. The following steps will walk you through creating an area for each grouping, as well as the controllers and views for each.
+
+#. In Solution Explorer, right-click the project name and create a new folder named ``Areas``.
+#. Right-click the ``Areas`` folder and create a new folder named ``Products``. This folder will house the controllers and views folders for the ``Products`` area.
+#. Right-click the ``Areas/Products`` folder and create a new folder named ``Controllers``. This folder will house all the controllers for the ``Products`` area.
+#. Right-click the ``Areas/Products/Controllers`` folder and add a new MVC controller named HomeController. By default, this controller will be created with a single action - ``Index``.
+#. Right-click the ``Areas/Products`` folder and create a new folder named ``Views``. This folder will house all the views for the ``Products`` area.
+#. Right-click the ``Areas/Products/Views`` folder and create a new folder named ``Home`` to match the controller name.
+#. Right-click the ``Areas/Products/Views/Home`` folder and add a new MVC view with the same name as the ``Home`` action: ``Index``.
+
+At this point, you should have a project folder hierarchy that includes the following:
+
+- Project name
+
+  - Areas
+
+    - Products
+
+      - Controllers
+
+        - HomeController
+
+      - Views
+
+        - Home
+
+          - Index.cshtml
+
+To add the Services and Deals areas, you would simply repeat steps 2-7, replacing "Products" with the name of the area you're defining.
+
+Once you've defined the folder hierarchy, you need to tell MVC that each controller is associated with an area. You do that by decorating the controller name with the ``[Area]`` attribute. Here's what the boiler-plate ``Products/HomeController`` looks like with the attribute added.
+
+.. code-block:: c#
+  :emphasize-lines: 4
+
+  ...
+  namespace MyStore.Areas.Products.Controllers
+  {
+      [Area("Products")]
```

Comment:
Right, putting the code for the controllers in a folder along side the corresponding views is a good practice, but it is not required by the runtime. It's the `[Area]` attribute on the controller that determines which area it is in.